### ISSUE TITLE
Ui.getCurrent().navigate() does not change Browser URL

### ISSUE THREAD

**Issue by mshabarov** [vaadin/flow#20478]
### Description of the bug

Based on provided forum post https://vaadin.com/forum/t/ui-getcurrent-navigate-does-not-change-browser-url/167788/1.

> Hi. After upgrading from 24.4.12 to 24.5.4 i have the problem, that the Browser URL does not change when UI.getCurrent().navigate() is called. The content and page title are set but the url stays the same.
Eg. i am currently on a view with route “/alldashboards”. There is a grid and on item click UI.getCurrent().navigate(“dashboard/” + item.getId()) is called. Navigation itself seems to work, since title and content are set, but the url stays “/alldashboards”.
Disabling the react router in the application properties did in fact solve the problem. We had the problem with any call of navigate(). Calling navigateToClient() instead worked fine, but since this needs a lot of code changes, we stick with disabling the react router.

To be investigated and reproduced.

### Expected behavior

URL should change.

### Minimal reproducible example

No.
A first try could be to run 24.4.12 app, upgrade to 24.5.4, then run it without any cleanup.

### Versions

- Vaadin / Flow version: 24.5.4

**Comment by mcollovati** [comment]
Tested on Flow starter project, upgrading from Vaadin 24.4.12 to 24.5.4 and 24.5.5.
Works as expected, the URL in the browser changes on navigation.

```java
@Route("alldashboards")
@AnonymousAllowed
public class AllDashboards extends VerticalLayout {

    public AllDashboards() {

        Grid<Integer> grid = new Grid<>();
        grid.addColumn(id -> "Dashboard " + id).setHeader("ID");
        grid.addItemClickListener(ev -> {
            UI.getCurrent().navigate("dashboard/" + ev.getItem());
        });
        grid.setItems(IntStream.rangeClosed(1, 10).boxed().collect(Collectors.toList()));
        add(grid);
    }
}
@Route("dashboard")
public class Dashboard extends Div implements HasUrlParameter<Integer> {

    private final Span text = new Span();

    public Dashboard() {
        add(text);
        add(new RouterLink("All dashboards", AllDashboards.class));
    }

    @Override
    public void setParameter(BeforeEvent beforeEvent, Integer integer) {
        text.setText("Welcome to dashboard " + integer.toString());
    }
}
```

**Comment by mcollovati** [comment]
Nope, clone the repo, added sample code, set version to 24.4.12, started the project and checked navigation worked, then bumped to 24.5.5, started and checked again.

**Comment by mcollovati** [comment]
Will try with a project from start.vaadin.com

**Comment by mcollovati** [comment]
Tested upgrade also with a custom bundle, but I'm still not able to reproduce

**Comment by mshabarov** [comment]
Tried with the https://github.com/vaadin/crm-tutorial and the following codes:

**ListView**
```java 
 List<String> items = Lists.newArrayList("one", "two", "three", "four");
Grid<String> grid = new Grid<>();
grid.setItems(items);
grid.addColumn(item -> item).setHeader("Items");
grid.addItemClickListener(event -> {
    String item = event.getItem();
    UI.getCurrent().navigate("/dashboard/" + item);
});
add(grid);
```
**DashboardView**
```java
public class DashboardView extends VerticalLayout implements HasUrlParameter<String> {
    @Override
    public void setParameter(BeforeEvent event, @OptionalParameter String parameter) {
        if (parameter == null) {
            add(new Div("Welcome anonymous."));
        } else {
            add(new Div(String.format("Welcome %s.", parameter)));
        }
    }
}
```

Couldn't reproduce, not with bundle, nor with Vite dev server.

**Comment by mshabarov** [comment]
If someone has a reliable reproducible steps please comment here, else I'm closing this ticket.

**Comment by vojtapol** [comment]
I have the same issue. For me, the browser address bar gets updated in production mode but not in development mode. I don't have a reproducible example, unfortunately. But I was wondering if the cases above were tested in development mode as well?

**Comment by mcollovati** [comment]
I tested it in development mode. Can you please check if you have a custom dev bundle in src/main/bundles folder? If so, try deleting it and restart the application

**Comment by vojtapol** [comment]
I just tried running production mode locally and it was still not updating the address bar. It does work correctly in production mode in the remote deployed environment. I wonder what differences could be causing this.

Could someone please add instructions on how to disable the React router?

**Comment by vojtapol** [comment]
I can confirm that opting out of React fixes my problem as well. Specifically I added this to my Spring properties:
```
vaadin.react.enable=false
```
So I think there is a bug somewhere but unfortunately I don't have time to reduce my giant project to a minimal reproducible example.

**Comment by mcollovati** [comment]
Did you perhaps checked and tried to clean the src/main/bundles folder before switching to Vaadin router?

**Comment by mshabarov** [comment]
Re-tested in production mode, but always got updated browser URL. I didn't clean the bundles folder.

**Comment by mgabor3141** [comment]
Had the same issue. Deleting the bundles and running vaadinClean and vaadinBuildFrontend fixed it.

**Comment by tbuchloh** [comment]
I had the same issue after upgrade from Vaadin 24.5.5 to 24.6.2.

The workaround worked for me:

`vaadin.react.enable=false`

**Comment by mstahv** [comment]
I wonder if we have some recent change that could have affected this issue? I started seeing this this in my app recently. Might be that something else (than version upgrade) I did lately contributed to the bundle so that react is again on even though I opted out 🤔 Need to investigate at some point.

Update while writing the report: The `vaadin.react.enable=false` in application.properties workaround seems to fix this for me as well. I had react disabled in the maven plugin, but that apparently is not enough, at least anymore.

**Comment by tepi** [comment]
Like others above, I'm unable to reproduce the issue, possibly it was flaky or not an issue for the latest version.

**Comment by mcollovati** [comment]
Not reproducible on current `main` (25.3): server-side `UI.navigate()` and `forwardTo()` both update the browser URL, in React-router and vaadin-router projects alike. The navigate-queue races behind this symptom were fixed right after the reported versions (#20446 and follow-ups, backported to 24.4/24.5/24.6 on 2024-11-25), so closing.
Please comment with your exact version if you still see it on 25.x.

**Comment by tachocarpintero** [comment]
The problem seems to be that navigate() works differently when it is used 

navigate(String url)

and 

navigate(clazz, new QueryParameters(...)) or navigate(String routeid, new QueryParameters(...))
